src/sleep/sleep.c

```diff
@@ -62,15 +62,31 @@ static int write_hibernate_location_info(void) {
                                        "Invalid hibernate type: %s", type);
 
         /* Only available in 4.17+ */
-        if (access("/sys/power/resume_offset", W_OK) < 0) {
+        if (access("/sys/power/resume_offset", R_OK) < 0) {
```

Comment:
there's no need to do the access() check if we open the file right-away here anyway. i mean, if the file doesn#t exist, then read_one_line_file() will return ENOENT, there's no point in checking that beforehand.

The reason the access() check was in place here is to avoid doing all the fiemap work on the raw device (which is security problematic and slow, and ...) if we don't have to.